### ISSUE TITLE
Getting linked Framework error while building appium-webdriveragent on Xcode 11.4 beta2

### ISSUE THREAD

**Issue by VenkateshPS** [appium/appium#13996]
## The problem

Getting linked Framework error while building appium-webdriveragent on Xcode 11.4 beta2

**error:** Building for iOS, but the linked framework ‘CocoaAsyncSocket.framework’ was built for tvOS + tvOS Simulator. (in target ‘WebDriverAgentLib’ from project ‘WebDriverAgent’)

Xcode 11.4 beta 2
Appium version:1.16.0
iOS 13.3.1
MAC OS:10.15.3

## Environment

Xcode 11.4 beta 2
Appium version:1.16.0
iOS 13.3.1
MAC OS:10.15.3

## Details

cd /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent 
mchatakonda01-Macmini:appium-webdriveragent venkateshsingh$ xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'id=203ae8e053dd9766ed62075a2ce47effdfa0f43c' test
Command line invocation:
    /Applications/Xcode-beta.app/Contents/Developer/usr/bin/xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=203ae8e053dd9766ed62075a2ce47effdfa0f43c test

2020-02-27 17:45:13.052 xcodebuild[8148:63966]  DTDeviceKit: deviceType from 203ae8e053dd9766ed62075a2ce47effdfa0f43c was NULL
2020-02-27 17:45:13.161 xcodebuild[8148:63966]  DTDeviceKit: deviceType from 203ae8e053dd9766ed62075a2ce47effdfa0f43c was NULL
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description
**error: Building for iOS, but the linked framework 'CocoaAsyncSocket.framework' was built for tvOS + tvOS Simulator. (in target 'WebDriverAgentLib' from project 'WebDriverAgent')**

Test session results, code coverage, and logs:
	/Users/venkateshsingh/Library/Developer/Xcode/DerivedData/WebDriverAgent-ciegwgvxzxdrqthilmrmczmqvrgu/Logs/Test/Test-WebDriverAgentRunner-2020.02.27_17-45-14-+0530.xcresult

Testing failed:
	Building for iOS, but the linked framework 'CocoaAsyncSocket.framework' was built for tvOS + tvOS Simulator.
	Testing cancelled because the build failed.

** TEST FAILED **

## Link to Appium logs

Create a [GIST](https://gist.github.com) which is a paste of your _full_ Appium logs, and link them here.
Do _NOT_ paste your full Appium logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include Appium logs!

## Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

Please git clone https://github.com/appium/appium and from the `sample-code` directory, use one of your favourite languages and sample apps to reproduce the issue.

In case a similar scenario is missing in sample-code, please submit a PR with one of the sample apps provided.

**Comment by mykola-mokhnach** [comment]
Please provide the full log with xcodebuild logging enabled.
Is the issue reproducible in appium@beta?

**Comment by VenkateshPS** [comment]
I insalled appium  using command 
npm install -g appium

so it should be latest

**Comment by mykola-mokhnach** [comment]
Unfortunately it's not. This version contains shrink-wrapped versions of all depending modules

**Comment by imurchie** [comment]
Try
```
npm uninstall -g appium
npm install -g appium@beta
```

**Comment by VenkateshPS** [comment]
ok, I will try that

**Comment by mykola-mokhnach** [comment]
Also make sure you did the cleanup before rebuilding the source

**Comment by VenkateshPS** [comment]
No errors on beta.

**Comment by imurchie** [comment]
Excellent!

**Comment by KiratiBhuva** [comment]
I am getting the same error with same environment. @VenkateshPS Can you give me the link of setup steps you followed?

**Comment by VenkateshPS** [comment]
I just followed what @imurchie mentioned

npm uninstall -g appium
npm install -g appium@beta

**Comment by BogdanCotofan** [comment]
Hello,
I have the same problem and appium@beta didn't solve the issue.
Tried on
MAC OS: 10.15.4
Appium 1.15.1
Xcode 11.4
iOS 13.4.1

Any suggestions?

**Comment by domignace** [comment]
Same problem here, 

I'm getting farther using appium-desktop 1.17.0 prerelease instead of 1.15.1.

**Comment by baopn97** [comment]
Issue occured in v1.15.1

**Comment by passionwang** [comment]
I have the same issue. do you solve it?
Xcode 11.4 beta 2
MAC OS:10.15.3

**Comment by KazuCocoa** [comment]
I wondered if below steps also caused the same issue from clean environment.
It the issue happened, it is appreciated to fix the xcode project in appium/WebDriverAgent since my environment worked well...
If it had no issue, your node environment has an issue. Please uninstall old installed appium module once.

https://github.com/appium/WebDriverAgent

```
git clone git@github.com:appium/WebDriverAgent.git
cd WebDriverAgent
cd ./Scripts/bootstrap.sh
/Applications/Xcode-beta.app/Contents/Developer/usr/bin/xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=203ae8e053dd9766ed62075a2ce47effdfa0f43c test
```

(The xcodebuild part should fit to your environment.)

**Comment by manjusomashekar** [comment]
I downloaded Appium 1.18.2. On running with desired capabilities on iOS, it threw a Code 65 error. On debugging, I ended up with WDA issue while building it on XCode 12.0.2 i.e., "Getting linked Framework error while building appium-webdriveragent". It was a dead-end and I just uninstalled appium. Now that you have released 1.18.3, let me check what happens in this version. Well, the problem is with WDA. I found an article explaining the same.. https://developer.apple.com/forums/thread/128797

Any workaround to fix this?

**Comment by cupcakepanda89** [comment]
@manjusomashekar  you can try this 
https://stackoverflow.com/questions/63607158/xcode-12-building-for-ios-simulator-but-linking-in-object-file-built-for-ios 
![Screen Shot 2020-11-04 at 10 33 44 AM](https://user-images.githubusercontent.com/16522221/98420508-3852a900-203c-11eb-8d41-b44f3648356d.png)
